src/libs/NetworkConnection.ts

```diff
@@ -21,15 +21,26 @@ type ResponseJSON = {
 let callbackID = 0;
 const reconnectionCallbacks: Record<string, () => void> = {};
 
+let isServerUp = true;
+let wasServerDown = false;
+
 /**
  * Loop over all reconnection callbacks and fire each one
  */
 const triggerReconnectionCallbacks = throttle(
     (reason) => {
-        Log.info(`[NetworkConnection] Firing reconnection callbacks because ${reason}`);
-        Object.values(reconnectionCallbacks).forEach((callback) => {
-            callback();
-        });
+        let delay = 0;
+
+        if (wasServerDown && isServerUp) {
+            delay = Math.floor(Math.random() * 21000); // Random delay between 0 and 20 seconds
```

Comment:
```suggestion
            delay = Math.floor(Math.random() * 61000); // Random delay between 0 and 60 seconds
```

Some ReconnectApp calls can take several seconds, so thousands of them within 20 seconds still feels like too many to me.

Reply:
That's fine